tardis/utilities/executors/sshexecutor.py

```diff
@@ -31,10 +44,60 @@ async def probe_max_session(connection: asyncssh.SSHClientConnection):
     return sessions
 
 
+class MFASSHClient(SSHClient):
+    def __init__(self, *args, mfa_secrets, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._mfa_responses = {}
+        for mfa_secret in mfa_secrets:
+            self._mfa_responses[mfa_secret["prompt"].strip()] = pyotp.TOTP(
+                mfa_secret["secret"]
+            )
+
+    async def kbdint_auth_requested(self) -> MaybeAwait[Optional[str]]:
+        """
+        Keyboard-interactive authentication has been requested
+
+        This method should return a string containing a comma-separated
+        list of submethods that the server should use for
+        keyboard-interactive authentication. An empty string can be
+        returned to let the server pick the type of keyboard-interactive
+        authentication to perform.
+        """
+        return ""
+
+    async def kbdint_challenge_received(
+        self, name: str, instructions: str, lang: str, prompts: KbdIntPrompts
+    ) -> MaybeAwait[Optional[KbdIntResponse]]:
+        """
+        A keyboard-interactive auth challenge has been received
+
+        This method is called when the server sends a keyboard-interactive
+        authentication challenge.
+
+        The return value should be a list of strings of the same length
+        as the number of prompts provided if the challenge can be
+        answered, or `None` to indicate that some other form of
+        authentication should be attempted.
+        """
+        # prompts is of type Sequence[Tuple[str, bool]]
+        try:
+            return [self._mfa_responses[prompt[0].strip()].now() for prompt in prompts]
+        except KeyError as ke:
+            msg = f"Keyboard interactive authentication failed: Unexpected Prompt {ke}"
+            logger.error(msg)
+            raise TardisAuthError(msg) from ke
+
+
 @enable_yaml_load("!SSHExecutor")
+@yaml_tag(eager=True)
 class SSHExecutor(Executor):
     def __init__(self, **parameters):
-        self._parameters = parameters
+        self._parameters = dict(parameters)
```

Comment:
You don't need to force a `dict` here, the `**parameters` already provide that.
```suggestion
        self._parameters = parameters
```
(Funny thing is that the spec is not clear on that, but all common implementations and type checkers adhere to it.)

Reply:
Something learned again. Thanks for the tip.